core-models/src/main/scala/com/mesosphere/usi/core/models/template/SimpleRunTemplateFactory.scala

```diff
@@ -8,23 +8,75 @@ import scala.collection.JavaConverters._
 
 object SimpleRunTemplateFactory {
 
+  sealed trait Command {
+    def build(): Mesos.CommandInfo.Builder
+  }
+
+  /**
+    * Use `/bin/sh -c` to execute passed command.
+    *
+    * @param cmd The shell command that is executed.
+    */
+  case class Shell(cmd: String) extends Command {
+    def build(): Mesos.CommandInfo.Builder = {
+      Mesos.CommandInfo
+        .newBuilder()
+        .setShell(true)
+        .setValue(cmd)
+    }
+  }
+
+  /**
+    * Use the entrypoint of the Docker image to execute the arguments. If the argument list is empty `CMD` from the
+    * Docker image is called.
+    *
+    * @param args A list of arguments passed.
+    */
+  class DockerEntrypoint private (args: Seq[String]) extends Command {
+    def build(): Mesos.CommandInfo.Builder = {
+      val builder = Mesos.CommandInfo
+        .newBuilder()
+        .setShell(false)
+
+      args.zipWithIndex.foreach { case (arg, index) => builder.setArguments(index, arg) }
+
+      builder
+    }
+  }
+
+  object DockerEntrypoint {
+
+    /** Factory method for [[com.mesosphere.usi.core.models.template.SimpleRunTemplateFactory.DockerEntrypoint]]; Scala Api. */
+    def apply(args: Seq[String]) = new DockerEntrypoint(args)
+
+    /** Factory method for [[com.mesosphere.usi.core.models.template.SimpleRunTemplateFactory.DockerEntrypoint]]; Java Api. */
+    def create(args: java.util.List[String]) = new DockerEntrypoint(args.asScala)
```

Comment:
can we just use the java varargs here?

Reply:
I am not sure how one can define Java varargs in Scala.

Reply:
Found it